util-picasso/src/main/java/com/ticketmaster/servos/util/picasso/FadeInTarget.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    public void startTransitionToDrawable(@Nullable Drawable drawable) {
+        if (drawable == null) drawable = new EmptyDrawable();
+
+        if (fade) {
+            setDrawableAndStartTransition(drawable);
+        } else {
+            setDrawableAtCurrentIndex(drawable);
+        }
+    }
+
+    private void setDrawableAtCurrentIndex(Drawable drawable) {
+        setDrawableByLayerId(currentIndex, drawable);
+        invalidateSelf();
+    }
+
+    private void setDrawableAndStartTransition(Drawable drawable) {
+        setDrawableByLayerId(getIndexOfNextBitmap(), drawable);
+
+        if (currentIndex == 0) {
+            super.startTransition(transitionDuration);
+        } else {
+            super.reverseTransition(transitionDuration);
+        }
+        currentIndex = getIndexOfNextBitmap();
+    }
+
+    @Override
+    @Deprecated
+    /** This class uses different transition methods,
+     *  see startTransitionToBitmap or startTransitionToDrawable **/
+    public void startTransition(int durationMillis) {
+        throw new RuntimeException("Use startTransitionToBitmap or startTransitionToDrawable instead");
+    }
+
+    @Override
+    @Deprecated
+    /** This class uses different transition methods,
+     *  see startTransitionToBitmap or startTransitionToDrawable **/
+    public void reverseTransition(int duration) {
+        throw new RuntimeException("Use startTransitionToBitmap or startTransitionToDrawable instead");
+    }
+
+    @Override
+    public void resetTransition() {
+        // This will be a cut, nothing we can do without rewriting TransitionDrawable.
+        if (currentIndex == 1) {
+            // Put current drawable into zero position so super.resetTransition can do its thing.
+            Drawable currentDrawable = getDrawable(1);
+
+            setDrawableByLayerId(1, new EmptyDrawable());
+            setDrawableByLayerId(0, currentDrawable);
+
+            currentIndex = 0;
+        }
+        // Must call for internal state tracking.
+        super.resetTransition();
+    }
+
+    private int getIndexOfNextBitmap() {
+        return currentIndex == 0 ? 1 : 0;
+    }
+
+    // Util methods
+
+    /**
+     * @return a Drawable, this may not be a BitmapDrawable if the Bitmap is null.
+     */
+    private static Drawable createDrawableFromBitmap(@Nullable Bitmap bitmap) {
+        if (bitmap == null) {
+            return new EmptyDrawable();
+        } else {
+            Drawable drawable = new BitmapDrawable(Resources.getSystem(), bitmap);
+            drawable.setBounds(new Rect(0, 0, bitmap.getWidth(), bitmap.getHeight()));
+            return drawable;
+        }
+    }
+
+    private static class EmptyDrawable extends ColorDrawable {
+        public EmptyDrawable() {
+            super(Color.TRANSPARENT);
+        }
+    }
+
+    // Picasso Stuff.
+
+    @Override
+    public void onBitmapLoaded(Bitmap bitmap, LoadedFrom from) {
+        fade = from != LoadedFrom.MEMORY;
```

Comment:
Discussion:

Rather than keeping track of whether or not to fade in an instance variable, why not just add a boolean to `startTransitionToDrawable/Bitmap()` methods? I think it'd be a little cleaner and you wouldn't have to be worrying about keeping track of state as much.